epochStart/bootstrap/shardStorageHandler.go

```diff
@@ -604,75 +604,76 @@ func getEpochShardDataAndNeededMetaBlock(
 	return epochShardData, neededMeta, nil
 }
 
-func getMiniBlocksInfo(epochShardData data.EpochStartShardDataHandler, neededMeta *block.MetaBlock, shardID uint32) *miniBlockInfo {
-	mbInfo := &miniBlockInfo{
+func getMiniBlocksInfo(epochShardData data.EpochStartShardDataHandler, neededMeta *block.MetaBlock, shardID uint32) *miniBlocksInfo {
+	mbsInfo := &miniBlocksInfo{
 		miniBlockHashes:              make([][]byte, 0),
 		fullyProcessed:               make([]bool, 0),
 		indexOfLastTxProcessed:       make([]int32, 0),
 		pendingMiniBlocksMap:         make(map[string]struct{}),
 		pendingMiniBlocksPerShardMap: make(map[uint32][][]byte),
 	}
 
-	setMiniBlockInfoWithPendingMiniBlocks(epochShardData, mbInfo)
-	setMiniBlockInfoWithProcessedMiniBlocks(neededMeta, shardID, mbInfo)
+	setMiniBlocksInfoWithPendingMiniBlocks(epochShardData, mbsInfo)
+	setMiniBlocksInfoWithProcessedMiniBlocks(neededMeta, shardID, mbsInfo)
 
-	return mbInfo
+	return mbsInfo
 }
 
-func setMiniBlockInfoWithPendingMiniBlocks(epochShardData data.EpochStartShardDataHandler, mbInfo *miniBlockInfo) {
+func setMiniBlocksInfoWithPendingMiniBlocks(epochShardData data.EpochStartShardDataHandler, mbsInfo *miniBlocksInfo) {
 	for _, mbHeader := range epochShardData.GetPendingMiniBlockHeaderHandlers() {
-		log.Debug("shardStorageHandler.setMiniBlockInfoWithPendingMiniBlocks",
+		log.Debug("shardStorageHandler.setMiniBlocksInfoWithPendingMiniBlocks",
 			"mb hash", mbHeader.GetHash(),
 			"len(reserved)", len(mbHeader.GetReserved()),
 			"index of first tx processed", mbHeader.GetIndexOfFirstTxProcessed(),
 			"index of last tx processed", mbHeader.GetIndexOfLastTxProcessed(),
 		)
 
 		receiverShardID := mbHeader.GetReceiverShardID()
-		mbInfo.pendingMiniBlocksPerShardMap[receiverShardID] = append(mbInfo.pendingMiniBlocksPerShardMap[receiverShardID], mbHeader.GetHash())
-		mbInfo.pendingMiniBlocksMap[string(mbHeader.GetHash())] = struct{}{}
-
-		if mbHeader.GetIndexOfLastTxProcessed() > -1 {
-			mbInfo.miniBlockHashes = append(mbInfo.miniBlockHashes, mbHeader.GetHash())
-			mbInfo.fullyProcessed = append(mbInfo.fullyProcessed, false)
-			mbInfo.indexOfLastTxProcessed = append(mbInfo.indexOfLastTxProcessed, mbHeader.GetIndexOfLastTxProcessed())
+		mbsInfo.pendingMiniBlocksPerShardMap[receiverShardID] = append(mbsInfo.pendingMiniBlocksPerShardMap[receiverShardID], mbHeader.GetHash())
+		mbsInfo.pendingMiniBlocksMap[string(mbHeader.GetHash())] = struct{}{}
+
+		isPendingMiniBlockPartiallyExecuted := mbHeader.GetIndexOfLastTxProcessed() > -1 && mbHeader.GetIndexOfLastTxProcessed() < int32(mbHeader.GetTxCount())-1
```

Comment:
an unit test for the exact encountered scenario would have been great

Reply:
Done